src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -812,10 +821,17 @@ RCLConsensus::Adaptor::validate(
         validationTime = lastValidationTime_ + 1s;
     lastValidationTime_ = validationTime;
 
+    if(!validatorKeys_.keys)
+    {
+        JLOG(j_.warn()) << "RCLConsensus::Adaptor::validate: ValidatorKeys "
+                            "not set\n";
+        return;
+    }
```

Comment:
As above, since you know that `keys` is set at this point, it might make sense, improve readability a little, and maybe speed things up by an amount too small to measure to set a reference and use it for the rest of the function:
```
auto const& keys = *validatorKeys_.keys;
```